src/Surfnet/StepupSelfService/SelfServiceBundle/Controller/Registration/SmsController.php

```diff
@@ -45,9 +46,17 @@ public function sendChallengeAction(Request $request)
             /** @var SmsSecondFactorService $service */
             $service = $this->get('surfnet_stepup_self_service_self_service.service.sms_second_factor');
 
-            if ($service->sendChallenge($command)) {
+            try {
+                $smsSendingSucceeded = $service->sendChallenge($command);
+            } catch (TooManyChallengesRequestedException $e) {
+                $form->addError(new FormError('ss.prove_phone_possession.challenge_request_limit_reached'));
+
+                return ['form' => $form->createView()];
```

Comment:
Maybe also show how many SMS's they still can request? Might be worth a discussion at the least...